src/injected/element-based-view-model-creator.ts

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { CardSelectionViewData } from 'common/get-card-selection-view-data';
+import { CardSelectionStoreData } from 'common/types/store-data/card-selection-store-data';
+import { UnifiedResult, UnifiedRule } from 'common/types/store-data/unified-data-interface';
+import { AssessmentVisualizationInstance } from 'injected/frameCommunicators/html-element-axe-results-helper';
+import { GetDecoratedAxeNodeCallback } from 'injected/get-decorated-axe-node';
+import { find, includes } from 'lodash';
+import { DictionaryStringTo } from 'types/common-types';
+
+export interface CheckData {
+    // tslint:disable-next-line: no-reserved-keywords
+    any: FormattedCheckResult[];
+    none: FormattedCheckResult[];
+    all: FormattedCheckResult[];
+}
+
+export type GetElementBasedViewModelCallback = (
+    rules: UnifiedRule[],
+    results: UnifiedResult[],
+    cardSelectionData: CardSelectionStoreData,
+) => DictionaryStringTo<AssessmentVisualizationInstance>;
+
+export type GetHighlightedResultInstanceIdsCallback = (
+    cardSelectionData: CardSelectionStoreData,
```

Comment:
Did you mean for this to be CardSelectionViewData? That is the type with 'highlightedResultUids'.

Reply:
The "ElementBasedViewModelCreator" receives the callback that you created as a part of the constructor and then uses it when necessary. All this is, is an interface/type for your callback. Hope that makes sense.